types/codec.go

```diff
@@ -14,4 +14,7 @@ func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 // RegisterInterfaces registers the sdk message type.
 func RegisterInterfaces(registry types.InterfaceRegistry) {
 	registry.RegisterInterface("cosmos.base.v1beta1.Msg", (*Msg)(nil))
+	// the interface name for MsgRequest is ServiceMsg because this is most useful for clients
+	// to understand - it will be the way for clients to introspect on available Msg service methods
+	registry.RegisterInterface("cosmos.base.v1beta1.ServiceMsg", (*MsgRequest)(nil))
```

Comment:
I am not sure I understand this comment: how will `"cosmos.base.v1beta1.ServiceMsg"` be useful for clients?

For clients to introspect on available Msg service methods, they just need to reflect on the proto definitions.

Reply:
Using the interface reflection service